src/Surfnet/ServiceProviderDashboard/Application/Metadata/JsonGenerator.php

```diff
@@ -238,8 +238,10 @@ private function generateMetadataFields(Entity $entity)
             $metadata['AssertionConsumerService:0:Binding'] = $entity->getAcsBinding();
             $metadata['AssertionConsumerService:0:Location'] = $entity->getAcsLocation();
             $metadata['NameIDFormat'] = $entity->getNameIdFormat();
+            $metadata['coin:signature_method'] = 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256';
             $metadata = array_merge($metadata, $this->generateSecurityMetadata($entity));
         } else if ($entity->getProtocol() == Entity::TYPE_OPENID_CONNECT) {
+            $metadata['coin:signature_method'] = 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256';
```

Comment:
I think you can get away with calling the `generateSecurityMetadata` method for both protocol types. Added benefit of that is that the setting of the signature method remains 'responsibility' of that method.

Reply:
The `certData` is omitted in OIDC so therefore I don't want to push it to manage. However renamed the method to `generateSecurityMetadata` to reflect it's responsibility better.

Reply:
This is fine too. By calling the method you would have saved a couple of lines of code.